File: in_Java/0415 Add Strings.java
class Solution {
    public String addStrings(String num1, String num2) {
        StringBuilder sb = new StringBuilder();
        char[] a = num1.toCharArray();
        char[] b = num2.toCharArray();
        int over = 0;
        int i = a.length-1;
        int j = b.length-1;
        int x;
        int y;
        int cur;
        while(i >= 0 || j >= 0){
            x = i < 0? 0 : (a[i] - '0');
            y = j < 0? 0 : (b[j] - '0');
            cur = x + y + over;
            over = cur / 10;
            cur %= 10;
            sb.insert(0, String.valueOf(cur));
            i--;
            j--;
        }
        String res = sb.toString();
        if(over == 1) res = "1" + res;
        return res;
    }
}

File: in_Python/0415 Add Strings.py
class Solution(object):
    def addStrings(self, num1, num2):
        """
        :type num1: str
        :type num2: str
        :rtype: str
        """
        res = []
        i = len(num1) - 1
        j = len(num2) - 1
        c = 0
        while i >= 0 or j >= 0:
            if i >= 0:
                a = int(num1[i])
            else:
                a = 0
            if j >= 0:
                b = int(num2[j])
            else:
                b = 0
            s = a + b + c
            res.insert(0, s % 10)
            c = s / 10
            i -= 1
            j -= 1
        r = ""
        #print(res)
        for x in res:
            r = r + str(x)
        if c == 1:
            r = '1' + r
        return r
        